PUSH_TO_FILE.md

```diff
@@ -709,24 +710,69 @@ annotation-based configuration and/or push-to-file mode:
     - Modify the Deployment's spec for the app container so that the
       `command` entrypoint includes sourcing of a bash-formatted secrets file.
 
-### Using the Helper Script to Patch the deployment
-There is a script in the secrets-provider-for-k8s bin directory named 
-generate-annotation-upgrade-patch.sh that can be used to generate a patch file.
+## Troubleshooting
 
-The patch can be output to a file:
+This section describes how to troubleshoot common Secrets Provider for Kubernetes issues.
+
+### Enable logs
+
+To enable logs, add the debug parameter to the application deployment manifest.
+```
+annotations:
+    conjur.org/debug-logging  "true"
+```
+For details, see [Reference Table of Configuration Annotations](#reference-table-of-configuration-annotations)
+
+### Display logs
+
+To display the Secrets Provider logs in Kubernetes.
+
+- Go to the app namespace 
 ```
-bin/generate-annotation-upgrade-patch.sh --push-to-file \$DEPLOYMENT_NAME > patch.json
+kubectl config set-context --current --namespace=<namespace>
 ```
-Test patch against a live deployment:
+- Find the pod name
 ```
-kubectl patch deployment \$DEPLOYMENT_NAME --type json --patch-file patch.json --dry-run=server
+kubectl get pods
 ```
-Preview the new deployment:
+- Display the logs
 ```
-kubectl patch deployment \$DEPLOYMENT_NAME --type json --patch-file patch.json --dry-run=server --output yaml
+kubectl logs <pod-name> -c <init-container-name>
 ```
-Apply patch:
+To display the Secrets Provider logs in Openshift.
+
+- Go to the app namespace
 ```
-kubectl patch deployment \$DEPLOYMENT_NAME --type json --patch-file patch.json
+oc project <project-name>
 ```
+- Find the pod name
+```
+oc get pods
+```
+- Display the logs
+```
+oc logs <pod-name> -c <init-container-name>
+```
+
+### Common issues displayed in the logs and resolutions
+
+|  Issue      |       Error code   | Resolution  |
+| ----------- | ------------------ | ------ |
+| Secrets Provider for Kubernetes failed to update secrets in file mode | `CSPFK039E` | Check that the secret key pairs have been configured correctly.  |
+| Failed to open annotations file                                       | `CSPFK041E` | The annotations file cannot be opened by the Secrets provider. Check the volume mounts and downward API configuration.  |
+| Annotation 'x' does not accept value                                  | `CSPFK042E` | The annotation has a key and a value. Here the is value invalid. Check the value for the annotation that is displayed in the error message.     |
+| Annotation 'x' does not accept value                                  | `CSPFK043E` | This is similar to CSPFK042E.  Check the value for the annotation that is displayed in the error message.      |
```

Comment:
The Resolutions for these two are a little bit unclear. Maybe something like "This annotation requires a specific type for its value. The value provided is the wrong type."
And for the second one, "This annotation only accepts one of a specific set of values...."